pkg/service/rulesvc/update.go

```diff
@@ -17,9 +17,19 @@ type UpdateOpts struct {
 
 func (s *Service) UpdateRule(ctx context.Context, in *UpdateOpts) (*rule.AccessRule, error) {
 	clk := s.Clock
-	target, err := s.ProcessTarget(ctx, in.UpdateRequest.Target)
-	if err != nil {
-		return nil, err
+
+	var target rule.Target
+	var err error
+	if in.Rule.Target.TargetGroupID != "" {
```

Comment:
Same as above, replace if statement with a variable assignment and don't duplicate the function call